unicorn-core/src/main/scala/org/virtuslab/unicorn/repositories/IdRepositories.scala

```diff
@@ -49,91 +48,92 @@ protected[unicorn] trait IdRepositories[Underlying] {
     // format: ON
 
     protected def queryReturningId = query returning query.map(_.id)
 
     final val tableName = query.baseTableRow.tableName
 
-    private implicit val executionContext = db.executor.executionContext
-
     /**
      * Finds one element by id.
      *
      * @param id id of element
-     * @param session implicit session
      * @return Option(element)
      */
-    def findById(id: Id)(implicit session: Session): Future[Option[Entity]] = db.run(byIdQuery(id).result.headOption)
+    def findById(id: Id): DBIO[Option[Entity]] = byIdQuery(id).result.headOption
 
     /**
      * Clones element by id.
      *
      * @param id id of element to clone
-     * @param session implicit session
      * @return Option(id) of new element
      */
-    def copyAndSave(id: Id)(implicit session: Session): Future[Id] =
-      findById(id).flatMap {
-        case Some(elem) => db.run(queryReturningId += elem)
-        case None => Future.failed(new NoSuchElementException(s"Element with $id doesn't exist"))
-      } // FIXME move this future to DbAction, and move execution context to unicorn
+    def copyAndSave(id: Id)(implicit ec: ExecutionContext): DBIO[Id] =
+      for {
+        elem <- findById(id)
+        result <- elem match {
+          case None => DBIO.failed(new NoSuchElementException(s"Element with $id doesn't exist"))
+          case Some(elem) => (queryReturningId += (elem))
+        }
+      } yield result
 
     /**
      * Finds one element by id.
      *
      * @param id id of element
-     * @param session implicit session
-     * @return Option(element)
+     * @return element
      */
-    def findExistingById(id: Id)(implicit session: Session): Future[Entity] =
-      findById(id).flatMap {
-        case Some(elem) => Future.successful(elem)
-        case None => Future.failed(new scala.NoSuchElementException(s"Element with $id doesn't exist"))
+    def findExistingById(id: Id)(implicit ec: ExecutionContext): DBIO[Entity] =
+      findById(id).map {
+        case Some(elem) => elem
+        case None => throw new NoSuchFieldException(s"For id: $id in table: $tableName")
       }
 
     /**
      * Finds elements by given ids.
      *
      * @param ids ids of element
-     * @param session implicit session
      * @return Seq(element)
      */
-    def findByIds(ids: Seq[Id])(implicit session: Session): Future[Seq[Entity]] = db.run(byIdsQuery(ids).result)
+    def findByIds(ids: Seq[Id]): DBIO[Seq[Entity]] = byIdsQuery(ids).result
 
     /**
      * Deletes one element by id.
      *
      * @param id id of element
-     * @param session implicit session
      * @return number of deleted elements (0 or 1)
      */
-    def deleteById(id: Id)(implicit session: Session): Future[Int] = db.run(byIdQuery(id).delete).map {
-      _.ensuring(_ <= 1, "Delete by id removed more than one row")
-    }(db.executor.executionContext)
+    def deleteById(id: Id): DBIO[Int] = byIdQuery(id).delete
```

Comment:
It was just a sanity check, either DB or Slick will have to be entirely broken for that `ensuring` to trigger. I'm ok with removing it.